src/transpose.jl

```diff
@@ -32,18 +32,69 @@ Base.:(==)(A::LinearMap, B::TransposeMap)    = issymmetric(A) && B.lmap == A
 Base.:(==)(A::LinearMap, B::AdjointMap)      = ishermitian(A) && B.lmap == A
 
 # multiplication with vector
-A_mul_B!(y::AbstractVector, A::TransposeMap, x::AbstractVector) =
-    issymmetric(A.lmap) ? A_mul_B!(y, A.lmap, x) : At_mul_B!(y, A.lmap, x)
+Base.@propagate_inbounds function LinearAlgebra.mul!(y::AbstractVector, A::TransposeMap, x::AbstractVector)
+    issymmetric(A.lmap) ? mul!(y, A.lmap, x) : error("transpose not implemented for $A")
```

Comment:
This throws an error, unless the map is symmetric, or a more specialized method for `TransposeMap{<:Any,<:SpecialMap}` is dispatched to. We could say here
```suggestion
    issymmetric(A.lmap) ? mul!(y, A.lmap, x) : mul!(y, transpose(A.lmap), x)
```
which would replace the pattern I mentioned above, that occurs in almost any `LinearMap`-type that we have. The only issue is with the `CustomMap`-type, which doesn't have an adjoint action defined. This type would get into an infinite loop here, and only eventually throw a `StackOverflowError`. To prevent this from happening, we have this `error` here. Things would simplify greatly if we replaced this here, but what to do with the infinite loop...?